src/python/tod/sim_noise.py

```diff
@@ -21,29 +21,17 @@ class AnalyticNoise(Noise):
     minimum frequency, etc.
 
     Args:
-        detectors (list): list of detectors.
-        rate (dict): dictionary of sample rates in Hertz.
-        fmin (dict): dictionary of minimum frequencies for high pass
-        fknee (dict): dictionary of knee frequencies.
-        alpha (dict): dictionary of alpha exponents (positive, not negative!).
-        NET (dict): dictionary of detector NETs.
+        detectors (list): List of detectors.
+        rate (dict): Dictionary of sample rates in Hertz.
+        fmin (dict): Dictionary of minimum frequencies for high pass
+        fknee (dict): Dictionary of knee frequencies.
+        alpha (dict): Dictionary of alpha exponents (positive, not negative!).
+        NET (dict): Dictionary of detector NETs.
+
     """
 
-    def __init__(self, detectors=None, rate=None, fmin=None, fknee=None, alpha=None, NET=None):
-        if detectors is None:
-            raise RuntimeError("you must specify the detector list")
-        if rate is None:
-            raise RuntimeError("you must specify the sample rates")
-        if fmin is None:
-            raise RuntimeError("you must specify the frequencies for high pass")
-        if fknee is None:
-            raise RuntimeError("you must specify the knee frequencies")
-        if alpha is None:
-            raise RuntimeError("you must specify the exponents")
-        if NET is None:
-            raise RuntimeError("you must specify the NET")
-
-        self._detectors = detectors
+    def __init__(self, *, detectors, rate, fmin, fknee, alpha, NET):
```

Comment:
This is how we can remove a lot of boiler plate and still require user to set the keyword arguments. Optional keyword arguments are still supported by giving them a default value.